storage/ksql/src/main/java/io/apicurio/registry/storage/impl/ksql/KafkaSqlCoordinator.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2020 Red Hat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.registry.storage.impl.ksql;
+
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import io.apicurio.registry.types.RegistryException;
+
+/**
+ * Coordinates "write" responses across threads in the Kafka-SQL storage implementation.  Basically this is used
+ * to communicate between the Kafka consumer thread and the waiting HTTP/API thread, where the HTTP thread is
+ * waiting for an operation to be completed by the Kafka consumer thread.
+ *
+ * @author eric.wittmann@gmail.com
+ */
+@ApplicationScoped
+public class KafkaSqlCoordinator {
+
+    private static final Object NULL = new Object();
+    private Map<UUID, CountDownLatch> latches = new ConcurrentHashMap<>();
+    private Map<UUID, Object> returnValues = new ConcurrentHashMap<>();
+
+    /**
+     * Creates a UUID for a single operation.
+     */
+    public UUID createUUID() {
+        UUID uuid = UUID.randomUUID();
+        latches.put(uuid, new CountDownLatch(1));
+        return uuid;
+    }
+
+    /**
+     * Waits for a response to the operation with the given UUID. There is a countdown latch for each operation.  The
+     * caller waiting for the response will wait for the countdown to happen and then proceed.  We also remove
+     * the latch from the Map here since it's not needed anymore.
+     *
+     * @param uuid
+     * @throws InterruptedException
+     */
+    public Object waitForResponse(UUID uuid) {
+        // TODO timeout should be configurable
+        try {
+            latches.get(uuid).await(30, TimeUnit.SECONDS);
+
+            Object rval = returnValues.remove(uuid);
+            if (rval == NULL) {
+                return null;
+            } else if (rval instanceof RegistryException) {
+                throw (RegistryException) rval;
+            }
+            return rval;
+        } catch (InterruptedException e) {
+          throw new RegistryException(e);
```

Comment:
Maybe a more concrete error message here?

Reply:
Good idea.  I'll do this.